amivapi/ldap.py

```diff
@@ -129,22 +133,15 @@ def _filter_data(data):
     res['gender'] = \
         u"male" if int(data['swissEduPersonGender']) == 1 else u"female"
 
-    if 'D-ITET' in data['ou']:
-        res['department'] = u"itet"
-    elif 'D-MAVT' in data['ou']:
-        res['department'] = u"mavt"
-    else:
-        res['department'] = u"other"
-
-    # ou contains all 'organization units'. This contains fields of study.
-    # Check if it contains any field of study assigned to us
-    ou_list = current_app.config['LDAP_MEMBER_OU_LIST']
-    is_member = bool(set(data['ou']).intersection(set(ou_list)))
-
-    if ('VSETH Mitglied' in data['ou']) and is_member:
-        res['membership'] = u"regular"
-    else:
-        res['membership'] = u"none"
+    # The departmentNumber field contains certain phrases for each department
```

Comment:
While I find the generator-style search very elegant, I think a bit further explanation on what the departmentNumber is and how this search works would be helpful to prevent wrong maintainance changes in the future to this very crucial part of the LDAP synchronisation.

Reply:
Good point, I think I will simply add a concrete example for the departmentNumber field to avoid confusion.